### PR TITLE
Revert some changes to rust compiler code (#42)

### PR DESCRIPTION
**Pull request by celinval**
### Description of changes: 

We would like to eventually take rustc as is. This change reverts minor changes we have done to the compiler overtime and replace them by calls to public APIs.

### Resolved issues:

None. There are still a few changes that we need to revert.

### Testing:

* How is this change tested?

* Is this a refactor change?

### Checklist
- [ ] Each commit message has a non-empty body, explaining why the change was made
- [ ] Methods or procedures are documented
- [ ] Regression or unit tests are included, or existing tests cover the modified code
- [ ] My PR is restricted to a single feature or bugfix

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 and MIT licenses.

### CODE REVIEW

**Review by zhassan-aws — APPROVED** [review]
LGTM

File: compiler/rustc_codegen_rmc/src/codegen/operand.rs [hunk 1]
```diff
@@ -165,18 +183,26 @@ impl<'tcx> GotocCtx<'tcx> {
                     FloatTy::F64 => Expr::double_constant_from_bitpattern(s.to_u64().unwrap()),
                 }
             }
-            (Scalar::Int(ScalarInt { size: 0, .. }), ty::FnDef(d, substs)) => {
-                self.codegen_fndef(*d, substs, span)
+            (Scalar::Int(int), ty::FnDef(d, substs)) => {
+                if int.size() == Size::ZERO {
```
Comment (danielsn) [inline compiler/rustc_codegen_rmc/src/codegen/operand.rs:187]: Would it be clearer to make this an `assert!`

**Review by danielsn — CHANGES_REQUESTED** [review]
Looks great, other than one comment about mixed control-flow

File: compiler/rustc_codegen_rmc/src/codegen/operand.rs [hunk 2]
```diff
@@ -86,38 +84,54 @@ impl<'tcx> GotocCtx<'tcx> {
...
+                                    .map(|b| Expr::int_constant(*b, Type::unsigned_int(8)))
+                                    .collect();
+                                let len = vec_of_bytes.len();
+                                let array_expr = Expr::array_expr(
+                                    Type::unsigned_int(8).array_of(len),
+                                    vec_of_bytes,
+                                );
+                                let data_expr = array_expr.array_to_ptr();
+                                let len_expr = Expr::int_constant(len, Type::size_t());
+                                return slice_fat_ptr(
```
[the remark this answers is not in the document]
Comment (celinval) [reply compiler/rustc_codegen_rmc/src/codegen/operand.rs:122]: Thanks for catching that. Sometimes my brain goes back to the old C++ way of writing code. o.O

**Review by danielsn — APPROVED** [review]

**Pull request merged**